File: packages/core/ios/overlays/PolygonViewModule.swift
import ExpoModulesCore

public class PolygonViewModule: Module {
    public func definition() -> ModuleDefinition {
        Name("PolygonView")
        
        View(PolygonView.self) {
            Events("onPolygonPress")
            
            Prop("points") { (view: PolygonView, points: [[String: Double]]) in
                view.setPoints(points)
            }
            
            Prop("fillColor") { (view: PolygonView, color: String) in
                view.setFillColor(color)
            }
            
            Prop("strokeColor") { (view: PolygonView, color: String) in
                view.setStrokeColor(color)
            }
            
            Prop("strokeWidth") { (view: PolygonView, width: Double) in
                view.setStrokeWidth(Float(width))
            }
            
            OnViewDidUpdateProps { (view: PolygonView) in
                // 属性更新完成后，如果还没连接地图，尝试连接
                if !view.isMapConnected() {
                    // 查找父视图 ExpoGaodeMapView
                    var parent = view.superview
                    while parent != nil {
                        if let mapView = parent as? ExpoGaodeMapView {
                            view.setMap(mapView.mapView)
                            return
                        }
                        parent = parent?.superview
                    }
                }
            }
        }
    }
}

File: packages/navigation/android/src/main/java/expo/modules/gaodemap/map/overlays/PolygonViewModule.kt
package expo.modules.gaodemap.map.overlays

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * Polygon 视图 Module
 */
class PolygonViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("PolygonView")

    View(PolygonView::class) {
      Events("onPolygonPress", "onPolygonSimplified")
      
      Prop<List<Any>?>("points") { view: PolygonView, points ->
        view.setPoints(points)
      }
      
      Prop<String?>("fillColor") { view: PolygonView, color ->
        view.setFillColor(color)
      }
      
      Prop<String?>("strokeColor") { view: PolygonView, color ->
        view.setStrokeColor(color)
      }
      
      Prop<Float>("strokeWidth") { view: PolygonView, width ->
        view.setStrokeWidth(width)
      }

      Prop<Float>("zIndex") { view: PolygonView, zIndex ->
        view.setZIndex(zIndex)
      }

      Prop<Double>("simplificationTolerance") { view: PolygonView, tolerance ->
        view.setSimplificationTolerance(tolerance)
      }
    }
  }
}